salt/metalk8s/kubernetes/apiserver-proxy/files/apiserver-proxy.conf.j2

```diff
@@ -10,12 +10,16 @@ events {
   worker_connections 16384;
 }
 
-{%- set masters_request = salt['mine.get'](
-        tgt='I@metalk8s:nodes:*:roles:master',
-        fun='control_plane_ip',
-        tgt_type='compound',
-    )
+{%- if pillar.get('apiserver_ip') %}
+{%-     set masters_request = {'ip': pillar.apiserver_ip} %}
```

Comment:
Why not squash with the previous commit?

Reply:
In GitHub history, we get:

```
- scripts: Add `apiserver_ip` as argument for restore script
- salt: Add ability to give `apiserver_ip` from pillar (this commit)
```

My question was: why not keep them together? I don't see another reason to use this pillar value other than during the restore script, so keeping this as a single commit would help readers understand why it was introduced.

Reply:
IMO make sense to split them as we add the feature to being able to specify a custom "apiserver IP" from pillar and then we use this feature in the restore script maybe I should just change the commit message of the first one